lib/libc/mips/gen/_setjmp.S

```diff
@@ -115,29 +115,21 @@ LEAF(_setjmp)
 	INT_S	v0, (_JB_FPREG_FCSR * SZREG)(a0)
 #endif /* ! __mips_soft_float */
 #if defined(__CHERI__)
-	/* jmp_buf is 64-bit aligned, so we need to round it up. */
-	cgetbase	t0, $c3
-	cgetoffset	t1, $c3
-	daddu		t1, t0, t1
-	daddiu		t0, t1, ((_JB_CHERI_START * SZREG) + _MIPS_SZCAP/8 - SZREG)
-	li		t2, _MIPS_CAP_ALIGN_MASK
-	and		t2, t2, t0
-	dsubu		t1, t2, t1
-
-	csc		$c11, zero, (_JB_CHERI_C11 * _MIPS_SZCAP/8)($c3)
-	csc		$c12, zero, (_JB_CHERI_C12 * _MIPS_SZCAP/8)($c3)
-	csc		$c13, zero, (_JB_CHERI_C13 * _MIPS_SZCAP/8)($c3)
-	csc		$c14, zero, (_JB_CHERI_C14 * _MIPS_SZCAP/8)($c3)
-	csc		$c15, zero, (_JB_CHERI_C15 * _MIPS_SZCAP/8)($c3)
-	csc		$c16, zero, (_JB_CHERI_C16 * _MIPS_SZCAP/8)($c3)
-	csc		$c17, zero, (_JB_CHERI_C17 * _MIPS_SZCAP/8)($c3)
-	csc		$c18, zero, (_JB_CHERI_C18 * _MIPS_SZCAP/8)($c3)
-	csc		$c19, zero, (_JB_CHERI_C19 * _MIPS_SZCAP/8)($c3)
-	csc		$c20, zero, (_JB_CHERI_C20 * _MIPS_SZCAP/8)($c3)
-	csc		$c21, zero, (_JB_CHERI_C21 * _MIPS_SZCAP/8)($c3)
-	csc		$c22, zero, (_JB_CHERI_C22 * _MIPS_SZCAP/8)($c3)
-	csc		$c23, zero, (_JB_CHERI_C23 * _MIPS_SZCAP/8)($c3)
-	csc		$c24, zero, (_JB_CHERI_C24 * _MIPS_SZCAP/8)($c3)
+	cgetdefault	$c4
+	csc		$c11, a0, ((_JB_CHERI_START * SZREG) + _JB_CHERI_C11 * _MIPS_SZCAP/8)($c4)
```

Comment:
I wonder if we should have a 
`#define _JB_CHERI_OFFSET(creg) ((_JB_CHERI_START * SZREG) + _JB_CHERI_##creg * _MIPS_SZCAP/8)`

Reply:
That would be cleaner looking and probably a good ideal.